10-JavaScript/dom-manipulation-live-code-notes.md

```diff
@@ -0,0 +1,136 @@
+# DOM Manipulation the Old Way: Live-code Notes
+
+## Running JavaScript on Websites
+
+Obviously we don't expect users to type out all their own JavaScript by hand. How do we get a website to connect to a JavaScript file? We'll add a link to our JavaScript file in our HTML, similar to the way we included CSS before.
+
+When our user goes to a website using their browser, the browser will make a request for an HTML page. This HTML page contains references to our CSS and JavaScript files. Then, these files will be loaded into the browser.
+
+1. Create a new directory called `browser-js`, with two files: `index.html` and `index.js`:
+
+1. Add this basic HTML to `index.html`:
+
+    ```html
+    <!-- index.html -->
+    <!DOCTYPE html>
+    <html>
+      <head>
+        <meta charset="utf-8">
+        <title>JavaScript Test Page</title>
+      </head>
+      <body>
+        <h1>Test page for JavaScript in the Browser</h1>
+
+        <div id="js-lecture-target">This is text that lives inside a div element with an id "js-lecture-target"!</div>
+      </body>
+    </html>
+    ```
+
+1. And finally, add this JavaScript to `index.js`:
+
+    ```javascript
+    // index.js
+    console.log('This is a test');
+    ```
+
+1. Modify your HTML page `index.html` by adding this `<script>` tag at the end of `<body>`, RIGHT before the closing `</body>` tag:
+
+    ```html
+    <!-- index.html -->
+    ...
+    <body>
+      ...
+
+      <script src="index.js" type="text/javascript"></script>
+    </body>
+    ```
+
+1. Ensure all files are saved
+1. Open up the Chrome Dev Tools and switch to the Console tab
+1. Reload the page (refresh the page)
+1. Observe that the test text `This is a test` from the JavaScript prints out!
+
+There are a lot of ways to load, then run JS scripts from a static webpage:
+`<script>` in head
+`<script>` in body
+
+Where Should `<script>` Go?
+
+Believe it or not, people have a lot of opinions on this topic.
+
+<details>
+
+  <summary>
+    In this class we'll always load our scripts at the end of the body. If you'd like to learn more, click here to expand and see our thoughts.
+  </summary>
+
+The reason why JS loading is so contentious is, when the browser encounters a `<script>` tag, it stops loading the HTML document. It goes and downloads your _entire_ script, which might be quite large and hosted halfway across the internet. Only once the script has finished downloading does it continue rendering the page. That means if you put your `<script>` tags before your content, the user gets to look at an empty white screen for a while while your scripts load. Not a great user experience.
```

Comment:
Consider adding a note that you might want to do this if your whole page is generated by JavaScript.